### DISCUSSION TITLE
Problem with FLSUN ADXL in Super Pad/Super Racer

### DISCUSSION THREAD

**Discussion by V12HA** [Guilouz/Klipper-Flsun-Speeder-Pad#160] [General]
---Edit---
Sorry about the typo in the title. It should have been 
"Problem with FLSUN ADXL in Speeder Pad/Super Racer"
---Edit---

After  re-installing the Speeder Pad, based to this: 
https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki
and this:
https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/wiki/Measure-resonances-with-FLSUN-ADXL , I get the following error:

MCU Protocol error

This is frequently caused by running an older version of the
firmware on the MCU(s). Fix by recompiling and flashing the
firmware.

Your Klipper version is: v0.12.0-114-ga77d0790
MCU(s) which should be updated:
adxl: Current version v0.10.0-455-ge002f0cf
Up-to-date MCU(s):
mcu: Current version v0.12.0-114-ga77d0790

It did work previously OK.

Any idea what to do to go on?

**Comment by Guilouz** [comment]
MCU of Flsun ADXL can't be updated so it is made unusable with the latest updates of klipper.

**Reply by V12HA** [reply]
Ok

**Reply by sgparry** [reply]
@Guilouz Please can you give more technical information on this? Is the MCU code read only? Or a lack of public source code for it?

**Reply by lin-ycv** [reply]
@sgparry I think the issue is that there's no way to enter dfu mode

**Reply by sgparry** [reply]
I have opened the case - the MCU is a Gigadevice GD32F303xx Cortex M4. The manual is here : [manual](https://www.gigadevice.com.cn/Public/Uploads/uploadfile/files/20230228/GD32F30x_User_Manual_EN_Rev2.9.pdf) It has four pads (ground, clk, dio, vcc) for what I presume is "old school" flash programming. I am hoping that pulling the two boot pins mentioned on page 39 to the correct values, the device might accept programming either via the pads or the USB. *gulp*

**Reply by sgparry** [reply]
Hmm, the GD32F303 has only UART programming. Why?? FLSUN, are we truly still stuck in the 1990s?

**Reply by lin-ycv** [reply]
> only UART programming

Question, what difference does uart make, why can't we flash if it only supports uart?

**Comment by V12HA** [comment]
What do you think, is it possible to use the 'old values' from the backup of the printer.cfg file from the previous installation?

Another possibility could be:
1. Install version 1.4 to Speeder Pad.
2. Do the measuring with the FLSUN ADXL
3. Take a note from the output
4. Install 1.2 version to Speeder Pad
5. Do the setup based to this version
6. Write the values to printer.cfg

Ok, it takes time, but is it a possibility...?

**Comment by Guilouz** [comment]
The resonance values ​​must be recalculated every time there is a change to the printer such as maintenance, belt, nozzle change or even if you move the printer.

So doing this every time I very much doubt that it will be very practical.

**Comment by brdi** [comment]
Tnx for the info
Just got myself a flsun adxl345 after uograding my V400 to the real Klipper :).

Flsun adxl does not work:(
So this is the reason. 
Bought and used another ADXL....

**Comment by sgparry** [comment]
I am sure it's possible, just non-trivial. The case has to be opened. The UART pads have no pins, so some kind of specialised cable and connector would need to be acquired / constructed. As for the software, I am clueless - it may need coding. Plus, one slip, these types of devices tend to brick. 

On 8 August 2024 02:05:19 BST, "Lin Yu-Chieh (Victor)" ***@***.***> wrote:
>> only UART programming
>
>Question, what difference does uart make, why can't we flash if it only supports uart? 
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad/discussions/160#discussioncomment-10269760
>You are receiving this because you were mentioned.
>
>Message ID: ***@***.***>
-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.
